realm/src/main/java/io/realm/Realm.java

```diff
@@ -194,10 +194,14 @@ public static Realm getInstance(Context context) {
      * model classes or version has has changed so a migration is required.
      */
     public static Realm getDefaultInstance() {
-        if (defaultConfiguration == null) {
+        final RealmConfiguration configuration;
+        synchronized (Realm.class) {
+            configuration = defaultConfiguration;
+        }
```

Comment:
I would say yes. Calling `Realm.setDefaultConfiguration()` should only be called from very few places: either in the `Application` on startup or some RealmHelper wrapper. Prevent calling `setDefaultConfigration()` twice would imply we need to add a `removeDefaultConfiguration()` as well, and TBH I cannot see any scenario where you would have a race condition on that method.